docs/docs/visual-testing-with-storybook.md

```diff
@@ -21,97 +21,200 @@ npx -p @storybook/cli sb init
+    config.module.rules[0].exclude = [/node_modules\/(?!(gatsby)\/)/]
+
+    // use installed babel-loader which is v8.0-beta (which is meant to work with @babel/core@7)
+    config.module.rules[0].use[0].loader = require.resolve("babel-loader")
+
+    // use @babel/preset-react for JSX and env (instead of staged presets)
+    config.module.rules[0].use[0].options.presets = [
+      require.resolve("@babel/preset-react"),
+      require.resolve("@babel/preset-env"),
+    ]
+
+    config.module.rules[0].use[0].options.plugins = [
+      // use @babel/plugin-proposal-class-properties for class arrow functions
+      require.resolve("@babel/plugin-proposal-class-properties"),
+      // use babel-plugin-remove-graphql-queries to remove static queries from components when rendering in storybook
+      require.resolve("babel-plugin-remove-graphql-queries"),
+    ]
+
+    // Prefer Gatsby ES6 entrypoint (module) over commonjs (main) entrypoint
+    config.resolve.mainFields = ["browser", "module", "main"]
+
+    // highlight-start
+    config.module.rules.push({
+      test: /\.(ts|tsx)$/,
+      loader: require.resolve("babel-loader"),
+      options: {
+        presets: [["react-app", { flow: false, typescript: true }]],
+        plugins: [
+          require.resolve("@babel/plugin-proposal-class-properties"),
+          // use babel-plugin-remove-graphql-queries to remove static queries from components when rendering in storybook
+          require.resolve("babel-plugin-remove-graphql-queries"),
+        ],
+      },
+    })
+
+    config.resolve.extensions.push(".ts", ".tsx")
+
+    // highlight-end
+    return config
+  },
 }
 ```
 
-> Note that if you're using a [StaticQuery](/docs/static-query/) in your components, `babel-plugin-remove-graphql-queries` is required to render them in Storybook. This is because the queries are run at build time in Gatsby, and will not have been run when rendering the components directly.
-
-> When using TypeScript, add this rule:
-
-```diff:title=.storybook/webpack.config.js
-// Prefer Gatsby ES6 entrypoint (module) over commonjs (main) entrypoint
-config.resolve.mainFields = ["browser", "module", "main"]
-+
-+   config.module.rules.push({
-+     test: /\.(ts|tsx)$/,
-+     loader: require.resolve('babel-loader'),
-+     options: {
-+       presets: [['react-app', {flow: false, typescript: true}]],
-+       plugins: [
-+         require.resolve('@babel/plugin-proposal-class-properties'),
-+         // use babel-plugin-remove-graphql-queries to remove static queries from components when rendering in storybook
-+         require.resolve('babel-plugin-remove-graphql-queries'),
-+       ],
-+     },
-+   });
-+
-+   config.resolve.extensions.push('.ts', '.tsx');
-+
-return config
+The `babel-preset-react-app` package will also need to be installed.
+
+```shell
+npm install --save-dev babel-preset-react-app
 ```
 
-**For Storybook v4:**
+### Storybook version 4
```

Comment:
```suggestion
With setup completed for Storybook 5, you can continue with the [information below to write stories](#writing-stories).

### Storybook version 4

To use Storybook version 4 with Gatsby, use the following setup instructions:
```

This section still jumps straight from a heading into source code, so even if it's getting removed later we should make it read better in the meantime.